TweetCell.m

```diff
@@ -21,4 +14,70 @@ - (void)setSelected:(BOOL)selected animated:(BOOL)animated {
     // Configure the view for the selected state
 }
 
+- (IBAction)didTapFavorite:(UIButton *)sender {
+    if (!self.tweet.favorited) {
```

Comment:
It worries me that favorited is checked but favorite count isn't. Does this need a check for negative numbers on favorites? Is there an upper bound?